nupic/research/frameworks/dendrites/routing/learn.py

```diff
@@ -259,18 +321,48 @@ def learn_to_route(
         print("{},{}".format(
             epoch, results["mean_abs_err"]
         ))
+        # track loss for plotting
+        if plot:
+            losses.append(results["mean_abs_err"])
+        # save models for future visualization or training
+        if save_interval and epoch % save_interval == 0:
+            if not os.path.exists(save_path):
+                os.makedirs(save_path)
+            torch.save(dendrite_layer.state_dict(),
+                       save_path + "dendrite_" + str(epoch))
+            if context_model:
+                torch.save(context_model.state_dict(),
+                           save_path + "context_" + str(epoch))
+
+    if plot:
+        import matplotlib.pyplot as plt
```

Comment:
It's generally bad practice to import in the code block. I suggest moving the import statement to the top of the file.